qutip/cy/inter.pyx

```diff
@@ -196,3 +196,53 @@ cdef complex _spline_complex_t_second(double x,
     cdef complex Mb = M[p] * dt2
     return te * (Mb * te * te + (y[p]   - Mb)) + \
            tb * (Me * tb * tb + (y[p+1] - Me))
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+@cython.cdivision(True)
+cdef double _step_float_cte(double x, double[::1] t, double[::1] y, int n_t):
+    if x < t[0]:
+        return y[0]
+    elif x >= t[n_t-1]:
+        return y[n_t-1]
+    cdef int ind = <int>(x/t[n_t-1]*(n_t-1))
```

Comment:
This will only work if tlist[0]=0, there was a case where this was not so recently and spline function no longer expect tlist to start at 0. Could you make it so tlist cans start from anywhere?